### ISSUE TITLE
Oh c’mon you guys...

### ISSUE THREAD

**Issue by kcrosley-leisurelabs** [openai/image-gpt#2]
Colab notebook?

(OpenAI’s dedication to crap documentation never ceases to amaze, amirite?)

**Comment by Kreijstal** [comment]
I'd like a colab notebook too but it's probably too much to ask.

**Comment by JonathanFly** [comment]
Here you go. Just does sampling, I didn't implement image completion.

https://gist.github.com/JonathanFly/eb61f0d31680e1b890f3a53fbaf31384

**Comment by methescripter** [comment]
@JonathanFly on the codelab How can I pass images as input? thanks

**Comment by JonathanFly** [comment]
> @JonathanFly on the codelab How can I pass images as input? thanks

That wasn't built in to the code released. It shouldn't be too hard to add but someone will have to do it. They do provide an encoding function for their cluster palette. I'll probably add it if I play more with this in the future.

**Comment by Kreijstal** [comment]
backlink https://github.com/huggingface/transformers/issues/5088

**Comment by apeguero1** [comment]
Hey guys, I modified @JonathanFly 's notebook to support conditional input such as cropped images.

[colab](https://colab.research.google.com/github/apeguero1/image-gpt/blob/master/Image_GPT_Sample_with_Conditioning.ipynb)

**Comment by JonathanFly** [comment]
Thanks @apeguero1 you improved it a lot.

**Comment by apeguero1** [comment]
No problem @JonathanFly glad to help (:

**Comment by dryn27** [comment]
@apeguero1 I tried Runtime>Run All, but it doesn't work. GPU is not the problem, and I cannot understand what the problem is. On "Functions from run.py", on "import tensorflow as tf", it shows an error and it says:

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.

**Comment by apeguero1** [comment]
Thanks for pointing this out @dryn27. Somehow bumping up to tensorflow version 1.15.0 fixes the import error. Just updated the colab.

**Comment by Randy-H0** [comment]
> Hey guys, I modified @JonathanFly 's notebook to support conditional input such as cropped images.
> 
> [colab](https://colab.research.google.com/github/apeguero1/image-gpt/blob/master/Image_GPT_Sample_with_Conditioning.ipynb)

Hi, Is there any way to sample with a higher resolution?

**Comment by eyalbetzalel** [comment]
@apeguero1 

Hi! thanks for the colab with the conditional sampling. 

is it possible to generate samples from a specific class from ImageNet? 

Generating photos of cars for example.

**Comment by apeguero1** [comment]
@eyalbetzalel hmm... in theory you could just finetune the whole model on a subset of the images with a particular label. Haven't tried this yet though.

Alternatively, Maybe you could make an [adapter](https://github.com/Adapter-Hub/adapter-transformers) per class  starting with the HF model made [here](https://colab.research.google.com/github/apeguero1/image-gpt/blob/master/Transformers_Image_GPT.ipynb) which would seem to be more efficient.

Ideally, there would be a way to train a single slightly modified architecture of the model that just adds a learned class conditional embedding to the token and positional embeddings. That would be pretty cool.

**Comment by proudbatdan** [comment]
Why’d you email me idk who you are

**Comment by fsyed0203** [comment]
i got an email too

**Comment by apeguero1** [comment]
@proudbatdan you might be signed up to watch this repo for any activity. You can always disable this.
